Detectors/EMCAL/reconstruction/src/RawReaderMemory.cxx

```diff
@@ -117,9 +121,48 @@ void RawReaderMemory::nextPage(bool doResetPayload)
   }
   if (mCurrentPosition + RDHDecoder::getMemorySize(mRawHeader) > mRawMemoryBuffer.size()) {
     // Payload incomplete
+    // Set to offset to next or buffer size, whatever is smaller
+    if (mCurrentPosition + RDHDecoder::getOffsetToNext(mRawHeader) < mRawMemoryBuffer.size()) {
+      mCurrentPosition += RDHDecoder::getOffsetToNext(mRawHeader);
+      if (mCurrentPosition + 64 < mRawMemoryBuffer.size()) {
+        // check if the page continues with a RDH
+        try {
+          auto nextheader = decodeRawHeader(mRawMemoryBuffer.data() + mCurrentPosition);
+          if (RDHDecoder::getVersion(nextheader) != RDHDecoder::getVersion(mRawHeader)) {
```

Comment:
I would make sure that the header version itself is acceptable (i.e. 6 or 7 ?) and also would always check the headerSize word, it never changes. 
Why do you need nested trt/catch with rethrow?

Reply:
Both checks (header version and header size) are part of decodeRawHeader. 

In principle you are right concerning the rethrow, the same error will be thrown.